content/en/synthetics/guide/canvas-content-javascript.md

```diff
@@ -0,0 +1,310 @@
+---
+title: Asserting canvas content with JavaScript
+
+further_reading:
+    - link: '/synthetics/browser_tests'
+      tag: 'Documentation'
+      text: 'Learn about Browser Tests'
+---
+
+## Overview
+
+Canvas content is rendered as raw pixel data rather than as structured HTML elements. This means you cannot target elements within a canvas using traditional selectors like XPath or CSS, since there is no underlying DOM representation of the visual content. As a result, actions such as **click** or **hover** cannot interact with elements drawn inside a canvas.
+
+To address this, you can leverage [custom JavaScript assertions][1] to analyze the canvas' pixel data and verify that the expected content is present.
+
+## What is an HTML canvas?
+
+The [`canvas`][2] element is an HTML tag that provides a drawable region in the browser, allowing you to render dynamic graphics using JavaScript. It's commonly used for displaying visual content such as charts, graphs, image editing tools, and animations.
+
+Supported by all major modern browsers, the `<canvas>` element is highly versatile. When combined with JavaScript libraries such as `Chart.js` or `D3.js`, it can render interactive visualizations that respond to user actions like clicks or hovers, enabling features such as custom tooltips or highlighting data points.
```

Comment:
```suggestion
Supported by all major modern browsers, the `<canvas>` element is versatile. When combined with JavaScript libraries such as `Chart.js` or `D3.js`, it can render interactive visualizations that respond to user actions like clicks or hovers, enabling features such as custom tooltips or highlighting datapoints.
```